components/zoho_catalyst/app/zoho_catalyst.app.ts

```diff
@@ -1,13 +1,98 @@
 import { defineApp } from "@pipedream/types";
+import { axios } from "@pipedream/platform";
+import {
+  DetectObjectsParams,
+  ExtractTextParams,
+  HttpRequestParams, PerformFaceDetectionParams, PerformModerationParams, Project,
+} from "../common/types";
 
 export default defineApp({
   type: "app",
   app: "zoho_catalyst",
-  propDefinitions: {},
+  propDefinitions: {
+    projectId: {
+      type: "integer",
+      label: "Project ID",
+      description: "Select a **Project** or provide a custom *Project ID*.",
+      async options() {
+        const projects: Project[] = await this.listProjects();
+        return projects.map(({
+          project_name: label, id: value,
+        }) => ({
+          label,
+          value,
+        }));
+      },
+    },
+    imagePath: {
+      type: "string",
+      label: "Image Path",
+      description:
+        "A file path in the `/tmp` directory. [See the documentation on working with files.](https://pipedream.com/docs/code/nodejs/working-with-files/)",
+    },
+    mode: {
+      type: "string",
+      label: "Mode",
+      optional: true,
+      default: "advanced",
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    async _httpRequest({
+      $ = this,
+      headers,
+      ...args
+    }: HttpRequestParams): Promise<object> {
+      return axios($, {
+        baseURL: `https://${this.$auth.base_api_uri}/baas/v1`,
+        headers: {
+          ...headers,
+          "Authorization": `Zoho-oauthtoken ${this.$auth.oauth_access_token}`,
+        },
+        ...args,
+      });
+    },
+    async listProjects(): Promise<Project[]> {
+      const projects = await this._httpRequest({
+        url: "/project",
+      });
+      return projects.data;
+    },
+    async detectObjectsInImage({
```

Comment:
Thanks for the suggestions Luan, I'm going to merge this as-is because we usually keep request methods marked as async even if it's not required, although it's a convention that's definitely worth discussing!